sbol_utilities/component.py

```diff
@@ -809,20 +867,23 @@ def ligation(reactants:List[sbol3.Component], assembly_plan:sbol3.Component)-> L
             # calculate sequence
             composite_sequence_str = ""
             composite_name = ""
-            for part in composite:
-                composite_sequence_str = composite_sequence_str + part.sequences[0].lookup().elements[:-fusion_site_length] #needs a version for linear
+            #part_subcomponents = []
+            part_extract_subcomponents = []
+            for part_extract in composite:
+                composite_sequence_str = composite_sequence_str + part_extract.sequences[0].lookup().elements[:-fusion_site_length] #needs a version for linear
                 # create participations
-                part_subcomponent = sbol3.SubComponent(part) # LocalSubComponent??
-                # if not in assemblye plan?
-                assembly_plan.features.append(part_subcomponent)
-                part_participation = sbol3.Participation(roles=[sbol3.SBO_REACTANT], participant=part_subcomponent)
-                participations.append(part_participation)
-                composite_name = composite_name + part.name
+                part_extract_subcomponent = sbol3.SubComponent(part_extract) # LocalSubComponent??
+                part_extract_subcomponents.append(part_extract_subcomponent)
+                # if not in assembl plan?
```

Comment:
Why are you adding a bunch of commented out code? Don't add commented out code unless there is a TODO explaining when and why it will be uncommented.

Reply:
removed, I added a TODO to update it to work outside of an assebly plan, but should this work outside of an assembly plan?